src/ChargeDrift/ChargeDrift.jl

```diff
@@ -99,20 +99,11 @@ function _set_to_zero_vector!(v::Vector{CartesianVector{T}})::Nothing where {T <
 end
 
 function _add_fieldvector_drift!(step_vectors::Vector{CartesianVector{T}}, current_pos::Vector{CartesianPoint{T}}, done::Vector{Bool}, 
-    electric_field::Interpolations.Extrapolation{<:StaticVector{3}, 3}, det::SolidStateDetector{T}, ::Type{S})::Nothing where {T, S}
-    for n in eachindex(step_vectors)
-       if !done[n]
-           step_vectors[n] += get_velocity_vector(electric_field, _convert_point(current_pos[n], S))
-           done[n] = (step_vectors[n] == CartesianVector{T}(0,0,0))
-       end
-    end
-    nothing
-end
-function _add_fieldvector_drift_nokill_when_nofield!(step_vectors::Vector{CartesianVector{T}}, current_pos::Vector{CartesianPoint{T}}, done::Vector{Bool}, 
-    electric_field::Interpolations.Extrapolation{<:StaticVector{3}, 3}, det::SolidStateDetector{T}, ::Type{S})::Nothing where {T, S}
+    electric_field::Interpolations.Extrapolation{<:StaticVector{3}, 3}, det::SolidStateDetector{T}, ::Type{S}, end_drift_when_no_field::Bool)::Nothing where {T, S}

```

Comment:
If you give a default value for `end_drift_when_no_field` here (`true`), the new code would still behave like the old code when explicitly calling `_add_fieldvector_drift!` and would, thus, be non-breaking
```suggestion
    electric_field::Interpolations.Extrapolation{<:StaticVector{3}, 3}, det::SolidStateDetector{T}, ::Type{S}, end_drift_when_no_field::Bool = true)::Nothing where {T, S}
```

Reply:
Yes, the default behavior doesn't change. When we want to calculate pulses in the dead layer, we need to set `end_drift_when_no_field` to `false`.

Reply:
Right now, the function is modified such that it needs an additional argument `end_drift_when_no_field`.
You might wanna add that its default value is `true` here, so that people can still use their old code.

Reply:
Ah, you mean the direct call of `_add_fieldvector_drift!`. Sure!